src/AWS4AuthRequest.jl

```diff
@@ -90,10 +101,14 @@ function sign_aws4!(method::String,
     end
 
     # Sort and lowercase() Headers to produce canonical form...
-    canonical_headers = sort!(map(headers) do (k, v)
-        string(lowercase(k), ':', replace(strip(v), r"\s+" => " "))
-    end)
-    signed_headers = join(sort!(map(lowercase∘first, headers)), ";")
+    unique_headers = sort!(unique!(map(lowercase∘first, headers)))
+    canonical_headers = map(unique_headers) do k
+        # All values present for this key (case-insensitive) with whitespace normalized
+        vs = map(i->_normalize_ws(last(headers[i])),
+                 findall(x->lowercase(first(x)) == k, headers))
+        string(lowercase(k), ':', join(vs, ','))
```

Comment:
```suggestion
        string(k, ':', join(vs, ','))
```

`k` is alrady lowercase